twoliter/embedded/Makefile.toml

```diff
@@ -1412,6 +1421,42 @@ pubsys \
 '''
 ]
 
+[tasks.fetch-variant]
+script_runner = "bash"
+script = [
+'''
+set -e
+
+export PATH="${TWOLITER_TOOLS_DIR}:${PATH}"
+
+if [ -z "${FETCH_VARIANT_LOCAL_ROOT_JSON}" ]; then
+   fetch_variant_root="./fetch-variant-root.json"
+   echo "Fetching root.json..."
+   curl -s -L -C - "${FETCH_VARIANT_ROOT_JSON}" -o "${fetch_variant_root#./}"
+   sha512sum -c <<<"${FETCH_VARIANT_ROOT_JSON_SHA512}  ${fetch_variant_root#./}"
+else
+   fetch_variant_root="${FETCH_VARIANT_LOCAL_ROOT_JSON}"
+fi
+
+if [ "${FETCH_VARIANT_UNIFIED}" == "true" ]; then
+   tuftool download \
+      "${BUILDSYS_VARIANT_DIR}" \
+      --target-name "${BUILDSYS_NAME_FRIENDLY}.img.lz4" \
+      --root "${fetch_variant_root:?}" \
+      --metadata-url "${FETCH_VARIANT_METADATA_URL}" \
+      --targets-url "${FETCH_VARIANT_TARGETS_URL}"
+else
+   tuftool download \
+      "${BUILDSYS_VARIANT_DIR}" \
+      --target-name "${BUILDSYS_NAME_FRIENDLY}.img.lz4" \
+      --target-name "${BUILDSYS_NAME_FRIENDLY}-data.img.lz4" \
+      --root "${fetch_variant_root:?}" \
+      --metadata-url "${FETCH_VARIANT_METADATA_URL}" \
+      --targets-url "${FETCH_VARIANT_TARGETS_URL}"
+fi
```

Comment:
Why do you need the `-data.img` file? I'd expect you  could do all the operations on just the boot / root / hash part of the main `.img` file, whether or not it contains the data partition.

As a practical matter, you end up depending on the variant's manifest in order to correct do Secure Boot registration, and (for VMware) to pick the correct OVF template.

Reply:
> Why do you need the `-data.img` file?

Answering my own question - `-data.img` would be needed to carry out a publish operation after the resigning, so it's better to pull the artifact down at the same time (if needed).

But again, unified vs. split is in the variant's manifest, so you can just read it from there instead.